pkg/controller/blockdevice/blockdevice_controller.go

```diff
@@ -91,6 +92,24 @@ func (r *ReconcileBlockDevice) Reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, err
 	}
 
+	switch instance.Status.ClaimState {
+	case openebsv1alpha1.BlockDeviceReleased:
+		jobController := cleaner.NewJobController(r.client, request.Namespace)
+		cleanupTracker := &cleaner.CleanupStatusTracker{JobController: jobController}
+		bdCleaner := cleaner.NewCleaner(r.client, request.Namespace, cleanupTracker)
+		ok, err := bdCleaner.Clean(instance)
+		if err != nil {
+			reqLogger.Error(err, "error while cleaning")
+			break
+		}
+		if ok {
+			err := r.updateBDStatus(openebsv1alpha1.BlockDeviceUnclaimed, instance)
```

Comment:
once a BD status is updated, should the cleaner job be deleted -- the one that is in succeed state?

Reply:
its in the reverse order. only after deleting job, clean will return `true`. So after that we update the status of BlockDevice.